server.js

```diff
@@ -1232,16 +1259,18 @@ async function objectBeatSender(PORT, thisId, thisIp, oneTimeOnly = false, immed
         let glbPath = path.join(targetDir, 'target.glb');
         let tdtPath = path.join(targetDir, 'target.3dt');
         var fileList = [jpgPath, xmlPath, datPath, glbPath, tdtPath];
-        objects[thisId].tcs = await utilities.generateChecksums(objects, fileList);
-    }
-
-    // if no target files exist, checksum will be undefined, so mark with checksum 0 (anchors have this)
-    if (typeof objects[thisId].tcs === 'undefined') {
-        objects[thisId].tcs = 0;
+        const tcs = await utilities.generateChecksums(objects, fileList);
+        if (objects[thisId]) {
```

Comment:
Question for my better understanding of underlying execution model of JS. 
These extra checks whether an object exists (so there is a valid access to it later in the function) are necessary only when there is asynchronous call beforehand in the currently executed function. Although, the current execution waits due to await, there is a switch of context by JS engine to execute another function. There the object can be modified/deleted and once the await finishes, there is a problem in the current function.

When there is no async call in the function, there is just one check necessary. The context switch by JS engine can happen only after the current function is finished.

Reply:
Yes, that's correct